sdks/csharp/sdk/AgonesSDK.cs

```diff
@@ -57,11 +61,20 @@ public sealed class AgonesSDK : IAgonesSDK
 		{
 			_logger = logger;
 			RequestTimeoutSec = requestTimeoutSec;
-			cts = cancellationTokenSource ?? new CancellationTokenSource();
+			
+			if (cancellationTokenSource == null)

```

Comment:
A general note for this and other `null` checks. IIRC the semantics of checking null with equality/inequality operator vs using `is` can be slightly different. I'm not familiar with this code, so I'll leave this as something that you might want to check?

https://stackoverflow.com/questions/40676426/what-is-the-difference-between-x-is-null-and-x-null

Reply:
They both compile down to the same IL when there is no overloaded `==` operator. Generally, I'd say `==` should be used when null checking because if somebody defined an overloaded operator for their type, it's probably best to use it.